compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingInterfaces.td

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_DIALECT_ENCODING_INTERFACES
+#define IREE_DIALECT_ENCODING_INTERFACES
+
+include "iree/compiler/Dialect/Encoding/IR/EncodingBase.td"
+
+// The interface is a subset of LinalgStructuredInterface.
+def EncodingInterface : OpInterface<"EncodingOp"> {
```

Comment:
IMO, the encoding ops do not need to implement an interface. We also don't need to implement `DestinationStyleOpInterface` for encoding ops. The set_encoding and unset_encoding ops are very straight-forward. How about just drop the interface files?

side note: we might need interfaces for attributes, and we can add the file when needed. In the context, we will need AttrInterface, but not OpInterface. So I suggest to remove EncodingInterface.* for now.